Content.Client/_Moffstation/Paper/UI/ForgeSignatureBoundUserInterface.cs

```diff
@@ -0,0 +1,57 @@
+using Content.Shared._Moffstation.Paper.Components;
+using Content.Shared.CCVar;
+using Robust.Client.UserInterface;
+using Robust.Shared.Configuration;
+
+namespace Content.Client._Moffstation.Paper.UI;
+
+/// <summary>
+/// Initializes a <see cref="ForgeSignatureWindow"/> and updates it when new server messages are received.
+/// </summary>
+public sealed class ForgeSignatureBoundUserInterface : BoundUserInterface
+{
+    [Dependency] private readonly IEntityManager _entManager = default!;
+    [Dependency] private readonly IConfigurationManager _cfgManager = default!;
+
+    [ViewVariables]
+    private ForgeSignatureWindow? _window;
+
+    public ForgeSignatureBoundUserInterface(EntityUid owner, Enum uiKey) : base(owner, uiKey)
+    {
+    }
+
+    protected override void Open()
+    {
+        base.Open();
+
+        _window = this.CreateWindow<ForgeSignatureWindow>();
+
+        if (!_entManager.HasComponent<ForgeSignatureComponent>(Owner))
+            return;
+
+        _window.SetMaxSignatureLength(_cfgManager.GetCVar(CCVars.MaxNameLength));
```

Comment:
```suggestion
        if (!_entManager.HasComponent<ForgeSignatureComponent>(Owner))
            return;
            
        _window = this.CreateWindow<ForgeSignatureWindow>();
        _window.SetMaxSignatureLength(_cfgManager.GetCVar(CCVars.MaxNameLength));
```
imo the window shouldn't open if the associated component is missing.